### PR TITLE
New Mappings

### PR DESCRIPTION
**Pull request by HU90m**

### CODE REVIEW

File: data/top_config.toml
Comment (marnovandermaas) [inline data/top_config.toml:314]: Nit: Missing comma

File: rtl/fpga/top_sonata.sv
Comment (HU90m) [inline rtl/fpga/top_sonata.sv:260]:
```suggestion
  assign spi_board_cipo = &{appspi_d1, ethmac_cipo, microsd_dat0};
```

File: data/top_config.toml
Comment (marnovandermaas) [inline data/top_config.toml:3]: Should we put pmod c in between the 0 and 1 like was specified in the pin spec?

**Comment by elliotb-lowrisc** [comment]
Needs pmodc pin mapping XDC. I've prepared the changes:
```
diff --git a/data/pins_sonata.xdc b/data/pins_sonata.xdc
index 8c8189c..1e7f96c 100644
--- a/data/pins_sonata.xdc
+++ b/data/pins_sonata.xdc
@@ -106,6 +106,14 @@ set_property -dict { PACKAGE_PIN C15 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [
 set_property -dict { PACKAGE_PIN H16 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmod1[6]}]
 set_property -dict { PACKAGE_PIN G16 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmod1[7]}]
 
+## PMODC
+set_property -dict { PACKAGE_PIN K15 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmodc[1]}]
+set_property -dict { PACKAGE_PIN J15 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmodc[2]}]
+set_property -dict { PACKAGE_PIN K16 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmodc[3]}]
+set_property -dict { PACKAGE_PIN J18 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmodc[4]}]
+set_property -dict { PACKAGE_PIN F18 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmodc[5]}]
+set_property -dict { PACKAGE_PIN J17 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmodc[6]}]
+
 ## Status LEDs
 set_property -dict { PACKAGE_PIN K5  IOSTANDARD LVCMOS33 } [get_ports led_legacy]
 set_property -dict { PACKAGE_PIN L4  IOSTANDARD LVCMOS33 } [get_ports led_cheri]
```

Let me know if you'd like me to push them, or apply them yourself.

**Comment by elliotb-lowrisc** [comment]
```
diff --git a/data/pins_sonata.xdc b/data/pins_sonata.xdc
index 8c8189c..b7fa41f 100644
--- a/data/pins_sonata.xdc
+++ b/data/pins_sonata.xdc
@@ -93,6 +93,14 @@ set_property -dict { PACKAGE_PIN E17 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [
 set_property -dict { PACKAGE_PIN D17 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmod0[6]}]
 set_property -dict { PACKAGE_PIN K13 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmod0[7]}]
 
+## PMODC (pins between PMOD0 and PMOD1 on shared connector)
+set_property -dict { PACKAGE_PIN K15 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmodc[0]}] ;# PMODC_1 on schematic
+set_property -dict { PACKAGE_PIN J15 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmodc[1]}] ;# PMODC_2
+set_property -dict { PACKAGE_PIN K16 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmodc[2]}] ;# ...
+set_property -dict { PACKAGE_PIN J18 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmodc[3]}]
+set_property -dict { PACKAGE_PIN F18 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmodc[4]}]
+set_property -dict { PACKAGE_PIN J17 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmodc[5]}]
+
 ## PMOD1
 set_property -dict { PACKAGE_PIN B18 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmod1[0]}]
 set_property -dict { PACKAGE_PIN E16 IOSTANDARD LVCMOS33 IO_BUFFER_TYPE NONE } [get_ports {pmod1[1]}]
```

**Comment by AlexJones0** [comment]
Software changes are pushed now and should hopefully mean that this PR is now passing. Please in review pay special notice to the HW change introducing the PMOD SF3 DPI as (a) I am not a hardware engineer and (b) I think I have taken a bit of a shortcut in `sonata-system.sv`, not sure of a nicer way to do what I've done. Please also generally check for any obvious mistakes - I'm rather tired and tried to rush these changes to get this PR in.

File: rtl/system/sonata_system.sv
Comment (AlexJones0) [inline rtl/system/sonata_system.sv:61]: Oops, I typo'd here. Should be `spi1_cipo_i`

**Review by marnovandermaas — COMMENTED** [review]
Just had a read-through of Alex's software changes and they look good to me. The PR needs a bit of cleanup but after that it is ready to go. I'm happy to do the cleanup myself Monday morning.

**Comment by AlexJones0** [comment]
Can confirm that the latest version of this branch does pass the manual "all blocks" pinmux check procedure on a bitstream built and loaded on my FPGA.

File: rtl/system/sonata_system.sv
Comment (AlexJones0) [inline rtl/system/sonata_system.sv:1116]: Should this not be something like `{lcd_dc_o, lcd_rst_o, lcd_cs_o}` to match the pin mapping in the [diagram](https://github.com/lowRISC/sonata-system/blob/e523ec89d8cab3069aa8630cffd18f33f09d985b/doc/ip/pinmux/pin_mappings.svg)? I think `dc` and `rst` are the wrong way around, and the backlight is missing?

**Comment by GregAC** [comment]
Haven't time for a detailed review but this is looking good from my perspective. Thanks for all of your hard work on this!

File: rtl/system/sonata_system.sv
Comment (marnovandermaas) [inline rtl/system/sonata_system.sv:1174]: Nit: remove space before square bracket.

File: data/top_config.toml
Comment (marnovandermaas) [inline data/top_config.toml:1]: Is memory start for GPIO used anywhere? I think it may be wrong because the first 0x40 bytes are hardcoded and not in the toml.

File: dv/verilator/top_verilator.sv
Comment (marnovandermaas) [inline dv/verilator/top_verilator.sv:307]: We should probably remove this one as well right? We don't want software to depend on the microSD card detection input being here.

File: data/top_config.toml
Comment (HU90m) [reply data/top_config.toml:3]: I personally think it's neater to have it appended as an additional, non-standard thing, but no strong feelings either way. Also, what are you referring to when you say pin spec?

File: rtl/system/sonata_pkg.sv
Comment (AlexJones0) [inline rtl/system/sonata_pkg.sv:18]: I think this width needs to be 7 to match the new mapping diagram? Right now the LCD backlight is going to PWM5, which is also being muxed to other pins.

File: rtl/fpga/top_sonata.sv
Comment (marnovandermaas) [reply rtl/fpga/top_sonata.sv]: Do we need this change before merging?

File: rtl/system/sonata_pkg.sv
Comment (HU90m) [reply rtl/system/sonata_pkg.sv]: fixed

File: data/top_config.toml
Comment (marnovandermaas) [reply data/top_config.toml:3]: I'm happy with this current configuration. I think I was referring to the original pinmapping PR, but let's leave this as is for now.

File: data/top_config.toml
Comment (HU90m) [reply data/top_config.toml:1]: Yes, it's not used anywhere, but I don't really want to add a special case in the top config verifier

File: dv/verilator/top_verilator.sv
Comment (HU90m) [reply dv/verilator/top_verilator.sv:307]: This has been part of the original proposal https://github.com/lowRISC/sonata-system/pull/271 . Do have a better idea of where this should go?

File: data/top_config.toml
Comment (marnovandermaas) [reply data/top_config.toml:1]: Ok, I've left it in the toml but updated it to exclude the gpio_board.

File: dv/verilator/top_verilator.sv
Comment (GregAC) [reply dv/verilator/top_verilator.sv:307]: Connected to the PLIC to allow interrupt on connection? We could add interrupt functionality to GPIO to support this  (though I'd intentionally be keeping it feature light, at some point it'd make sense to just use the OT one instead...).

I think for now this is no problem, we'd probably want a dedicated 1-bit GPIO to enable better compartmentalization longer-term.

Comment (GregAC) [reply dv/verilator/top_verilator.sv:307]: Though I guess the PLIC can disable the interrupt and you can read the state of the interrupt so maybe direct PLIC connection (via debouncer) is actually the elegant solution, anyway definitely a post 1.0 thing.

Comment (marnovandermaas) [reply dv/verilator/top_verilator.sv]: I think we should remove it from the general GPIO before the 1.0 release and then figure out how to connect it properly (PLIC or 1-bit GPIO) after 1.0. We cannot remove it from the general GPIO post 1.0 because it would break software relying on the placement of this bit.

Comment (marnovandermaas) [reply dv/verilator/top_verilator.sv]: Nevermind, I just chatted to Alex and I think we can leave it in for now.

**Review by marnovandermaas — APPROVED** [review]
This PR now looks good to me. Let's wait for @AlexJones0 to approve and then @HU90m can merge when he thinks it's ready.

**Review by AlexJones0 — APPROVED** [review]
Giving my review as per @marnovandermaas's request.

This review is not a review of the commits of the PR itself, but just an approval that from the software side of thing everything appears to be working.

- As can be seen, this PR is passing on both FPGA and simulator in CI, and I have also run `test_runner` on an FPGA with a local bitstream build and checked that this works. 
- I have further ran the manual pinmux checker application and confirmed that we still have the ability to mux GPIO, UART, I2C & SPI devices on the same set of PMOD pins. 
- With the changes from [my local branch](https://github.com/alexjones0/sonata-software/tree/pin_mapping_changes), I am able to successfully run and interact with the `snake_demo` and `leds_and_lcds` / `sonata_simple_demo`. 
- I have manually tested changing the LCD backlight over a range of values and can confirm that on a physical board, this seems to vary the intensity of backlight as you would expect.
- I have used a multimeter to verify that the new PMODC GPIO can be interacted with as GPIO outputs, as a quick sanity check for PMODC.

It is worth noting that this PR does not fix all software - just enough to get the tests, manual pinmux check and bootloader running, so as to reduce the size of this PR (and limit it to HW changes critical for a 1.0 release). Software changes to fix the remainder of software should be introduced in follow up PRs.

**Comment by elliotb-lowrisc** [comment]
:tada: :tada: :tada:

**Pull request merged**
